torch/autograd/_functions/basic_ops.py

```diff
@@ -116,10 +116,30 @@ def sort_args(a, b, key=torch.is_tensor):
     return (a, b, True) if key(a) else (b, a, False)
 
 
+def gen_inputs(g, a, b):
+    tensor, constant, tensor_first = sort_args(a, b, key=is_node)
+    assert tensor.hasType()
+    type = str(tensor.type().scalarType())
+    broadcast = False
+    if len(tensor.type().sizes()) > 1:
+        broadcast = True
+    constant = g.constant(constant, [tensor.type().sizes()[-1]], type).typeAs(tensor)
```

Comment:
This is not what we want long term and it should be documented accordingly (with this code, if you add a constant n to a 100 x 100 tensor, ONNX will serialize a size 100 1-dim tensor; not great.)  In hopefully not too long, we'll serialize this as a scalar getting rid of this goofiness.